src/databricks/labs/ucx/hive_metastore/federation.py

```diff
@@ -31,58 +48,161 @@ def enable(self):
         self._installation.save(config)
 
 
-class HiveMetastoreFederation:
+class HiveMetastoreFederation(SecretsMixin):
     def __init__(
         self,
-        workspace_client: WorkspaceClient,
+        installation: Installation,
+        ws: WorkspaceClient,
         external_locations: ExternalLocations,
         workspace_info: WorkspaceInfo,
         enable_hms_federation: bool = False,
     ):
-        self._workspace_client = workspace_client
+        self._ws = ws
         self._external_locations = external_locations
         self._workspace_info = workspace_info
         self._enable_hms_federation = enable_hms_federation
+        self._installation = installation
 
-    def register_internal_hms_as_federated_catalog(self) -> CatalogInfo:
+    # Supported databases and version for HMS Federation
+    supported_db_vers: ClassVar[dict[str, list[str]]] = {
+        "mysql": ["2.3", "0.13"],
+    }
+
+    def create_from_cli(self, prompts: Prompts):
         if not self._enable_hms_federation:
             raise RuntimeWarning('Run `databricks labs ucx enable-hms-federation` to enable HMS Federation')
         name = self._workspace_info.current()
-        connection_info = self._get_or_create_connection(name)
+
+        ext_hms = None
+        try:
+            ext_hms = self._get_ext_hms()
+        except ValueError:
+            logger.info('Failed to retrieve external Hive Metastore connection information')
+
+        if ext_hms and prompts.confirm(
+            f'A supported external Hive Metastore connection was identified: {ext_hms.db_type}. Use this connection?'
+        ):
+            connection_info = self._get_or_create_ext_connection(name, ext_hms)
+        else:
+            connection_info = self._get_or_create_int_connection(name)
         assert connection_info.name is not None
+        return self._register_federated_catalog(connection_info)
+
+    def _get_ext_hms(self) -> ExtHms:
```

Comment:
I think parse would be more accurate here. I though you get a "connection", the external hive metastore here are the login details from the spark config

```suggestion
    def _parse_ext_hms(self) -> ExtHms:
```

Reply:
Sure